### PR TITLE
Create Section 12 'Custom Credentials' of the Controller UG

### PR DESCRIPTION
**Pull request by ianf77**
Added assembly and files.

Create Section 12 of the Controller user Guide

https://issues.redhat.com/browse/AAP-15823

### CODE REVIEW

File: downstream/assemblies/platform/assembly-controller-custom-credentials.adoc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+Custom credentials support the following ways of injecting their authentication information:
+
+* Environment variables
+* Ansible extra variables
+* File-based templating, that is, generating `.ini` or `.conf` files that contain credential values
+
+You can attach one SSH and multiple cloud credentials to a Job Template.
+Each cloud credential must be of a different type. 
+Only one AWS credential or one GCE credential, etc., are permitted.
```
Comment (EMcWhinn) [inline downstream/assemblies/platform/assembly-controller-custom-credentials.adoc:17]: Avoid the use of "etc.". Use "and so on" or maybe just "...one GCE credential is permitted."

File: downstream/modules/platform/proc-controller-create-credential-type.adoc [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  "label": "Password"
+   }
+  ],
+ "required": ["username", "password"]
+}
+----
++
+View more JSON examples at link:https://www.json.org/json-en.html[The JSON website]
++
+The configuration in JSON format below show each field and how they are used:
```
Comment (EMcWhinn) [inline downstream/modules/platform/proc-controller-create-credential-type.adoc:65]: Suggested re-write:

The following configuration in JSON format shows each field and how they are used:

File: downstream/modules/platform/proc-controller-create-credential-type.adoc [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+      "type": "string",
+      "choices": ["A", "B", "C"]
+  }]
+},
+----
+
+. In the *Injector Configuration* field, enter environment variables or extra variables that specify the values a credential type can inject.
+The format can be in YAML or JSON (see examples in the previous step).
++
+The configuration in JSON format below show each field and how they are used:
```
Comment (EMcWhinn) [inline downstream/modules/platform/proc-controller-create-credential-type.adoc:119]: Suggested re-write:

The following configuration in JSON format shows each field and how they are used:

File: downstream/modules/platform/proc-controller-create-credential-type.adoc [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+. Your newly created credential type appears on the list of credential types:
++
+image:credential-types-new-listed.png[New credential type]
+
+. Click the image:leftpencil.png[Edit,15,15] icon to modify the credential type options.
++
+[NOTE]
+====
+In the *Edit* screen, you can modify the details or delete the credential.
+If the *Delete* button is disabled, it is indication that the credential type is being used by a credential, and you must delete the credential type from all the credentials that use it before you can delete it.
```
Comment (EMcWhinn) [inline downstream/modules/platform/proc-controller-create-credential-type.adoc:204]: Avoid "button", suggested re-write "If the *Delete* option is disabled.." or " If *Delete* is disabled.."

File: downstream/modules/platform/ref-controller-api-considerations.adoc [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+| Network |S ets environment variables, for example `ANSIBLE_NET_AUTHORIZE`
+| SCM | Source Control
+| Cloud | EC2, AWS
+| | Lots of others
+| Insights |Insights
+| Galaxy | galaxy.ansible.com, console.redhat.com
+| | on-premise {HubName}
+|===
+
+= Content verification
```
Comment (EMcWhinn) [inline downstream/modules/platform/ref-controller-api-considerations.adoc:30]: Should this be ".Verification"?

Comment (ianf77) [reply downstream/modules/platform/ref-controller-api-considerations.adoc:30]: No. It's a separate section. It was too short to create a module just for it.

**Pull request merged**